### DISCUSSION TITLE
pyqt5在deepin 20.3上设置窗口透明不起作用

### DISCUSSION THREAD

**Discussion by myFirstway** [linuxdeepin/developer-center#3413] [Development Questions | 开发答疑] [answered]
在deepin 20.7上可以看到的效果是窗口透明可以看到窗口下的软件的内容，在deepin 20.3上以及20.3升级到的20.7上窗口是完全没有透明，代码如下:
![1662709879041](https://user-images.githubusercontent.com/7514693/189300466-f2afe806-c742-42d4-9b73-a7cf578edcbc.png)

![1662709942362](https://user-images.githubusercontent.com/7514693/189300473-51c9331b-bb79-4e27-891a-b3bb32159ac4.png)
`import sys
from PyQt5.QtCore import Qt
from PyQt5.QtWidgets import QApplication, QLabel,QWidget,QGraphicsOpacityEffect
class draw_select_win (QWidget):
  def __init__(self, *args, **kwargs):
    self.app = QApplication(sys.argv)
    super().__init__(*args, **kwargs)
    self.stop = False
    self.resize(80, 40) 
    # self.setWindowFlags(Qt.FramelessWindowHint)  # 窗口无边框
    self.setWindowOpacity(0.5)  # 设置窗口透明度   
    # self.setAttribute(Qt.WA_TranslucentBackground)  # 窗口透明
  
  def mouseRelease(self ,ev):
    self.stop = True
  
  def draw(self, x,y,width,height):
    print(x,y,width,height)
    window.move(x,y)
    window.resize(width,height)
    window.show()

  def show(self):
    global window
    window = QWidget()   
    window.move(0, 0)
    # window.setWindowFlags(Qt.FramelessWindowHint|Qt.WindowStaysOnTopHint|Qt.ToolTip)  # 窗口无边框
    # window.setWindowFlags(Qt.FramelessWindowHint|Qt.WindowStaysOnTopHint)  # 窗口无边框
    window.setStyleSheet("background-color:rgba(255,0,0,0.5);")
    window.setWindowOpacity(0.4)  # 设置窗口透明度   
    window.mousePressEvent = self.mouseRelease
    window.show()
    sys.exit(self.app.exec_())

label = draw_select_win()
label.show()`

**Comment by BLumia** [comment, marked as answer]
标题栏是窗口管理器的，如果你想标题栏那部分也透明的话，你得隐藏掉标题栏然后自己搞个假的标题栏。
以及窗口透明本身是需要高效模式（指开启窗口混成）的。

顺带，建议更新下系统哈。

**Reply by myFirstway** [reply]
arm版的机器上不能够开启窗口特效，这个怎么设置呢？

**Reply by BLumia** [reply]
需要显卡和窗管都支持。也可以考虑检索一下你所使用的环境使用其它窗口管理器（比如原版 kwin 之类）是否可以开启混成（compositing）。

**Reply by myFirstway** [reply]
> 

好的 谢谢啊
